src/librustc_expand/module.rs

```diff
@@ -280,10 +280,14 @@ pub fn default_submod_path<'a>(
                 mod_name,
             );
             err.help(&format!(
-                "to create the module `{}`, create file \"{}\"",
+                "to create the module `{}` here, create file \"{}\"",
                 mod_name,
                 default_path.display(),
             ));
+            err.help(&format!(
+                "if there is `mod {}` elsewhere in the crate already, import it with `use crate::` instead",
+                mod_name,
+            ));
```

Comment:
You can do the following instead
```rust
err.span_suggestion_verbose(
    span,
    "if the module exists elsewhere in the crate already, you can import instead",
    format!("use crate::{};", mod_name),
    Applicability::MaybeIncorrect,
)
```

That being said, it'd be nice to actually check that the module exists in the crate and provide the suggestion with higher certainty. Do not have the time to check what the code would have to be to verify that `crate::mod_name` exists, but @petrochenkov might know of the top of their head. Otherwise, would you mind exploring how to do that, @kornelski? We want to minimize the cases where we provide incorrect suggestions, as those can be worse than terse ones. If we can make that check, then we can provide the help to create the file exclusively if the `mod` doesn't exist alredy.

Reply:
I'm happy to dig in further, but I'd prefer to land this change anyway. It starts with a disclaimer, it intentionally only suggests `use` instead of showing a potentially-incorrect path. So I think risk of confusion is low, and mentioning `use` is an important improvement.